src/main/java/net/openhft/chronicle/wire/VanillaMethodWriterBuilder.java

```diff
@@ -56,6 +57,10 @@
             Observer.class
     );
 
+    private static final LongHashFunction HASH_FUNCTION = LongHashFunction.city_1_1();
```

Comment:
I didn't put a lot of thought into which hash to use, so happy to change to something else.

Reply:
It doesn't need to be a complete Hash, you can have the most important details being readable

Reply:
Yes, only the portion which is truncated is replaced by its hash. we keep as much of the interface list as we can and we keep all of `MetadataAware|Intercepting|WireType|etc.`. It's so we can safely truncate without introducing naming collisions for classes that only differ in their truncated portion.

Reply:
I replaced the zero-allocation-hashing city hash with the one from Maths, but it doesn't support hashing a substring of a StringBuilder so what you get now is the hash of the entire classname pre-truncation instead of the hash of just the bit that was truncated. That's fine though, all the original properties hold.

Reply:
You can see some examples of what it produces in [the test](https://github.com/OpenHFT/Chronicle-Wire/pull/644/files#diff-dda75cf859856e30c3a4e3004c0368cb1a23e447196272bfea3e75fdcfda0aa2)